src/wrapper.rs

```diff
@@ -482,6 +482,52 @@ impl DxcLibrary {
     }
 }
 
+#[derive(Debug)]
+pub struct DxilContainerReflection {
+    inner: ComPtr<IDxcContainerReflection>,
+}
+
+impl DxilContainerReflection {
+    pub fn new(inner: ComPtr<IDxcContainerReflection>) -> Self {
+        Self { inner }
+    }
+
+    pub fn load(&self, p_dxc_blob: &IDxcBlob) {
+        unsafe {
+            self.inner.load(p_dxc_blob);
+        }
+    }
+
+    pub fn find_first_part_kind(&self) -> u32 {
+        let mut shader_idx = 0u32;
+        unsafe {
+            self.inner.find_first_part_kind(
+                Self::four_cc('D' as u32, 'X' as u32, 'I' as u32, 'L' as u32),
+                &mut shader_idx,
+            );
+        }
+        shader_idx
+    }
+
+    fn four_cc(ch0: u32, ch1: u32, ch2: u32, ch3: u32) -> u32 {
+        0u32 | ch0 | ch1 << 8 | ch2 << 16 | ch3 << 24
+    }
+
+    pub fn get_part_reflection(
+        &self,
+        idx: u32,
+        p_reflection: *mut *mut ID3D12ShaderReflection,
```

Comment:
Normally we'd provide a safe wrapper around the type here, as well as safe initialization and returning the object (eg. `return_hr_wrapped` below).

Not sure if we want to make an exception for the time being until further examining how to model this in a safe wrapper, or leave the PR open until having done so?

Reply:
changed in latest commits.